### DISCUSSION TITLE
How to use rnx-bundle in an application created via the "react-native init" command?

### DISCUSSION THREAD

**Discussion by Thanaen** [microsoft/rnx-kit#1556] [Q&A] [answered]
Hi,

I am very interested in the features offered by rnx-bundle (type-safety and tree shaking integrated in Metro).
I tried to follow this guide:
https://microsoft.github.io/rnx-kit/docs/guides/bundling

But there is something I don't understand.
In the chapter "Configuring Your App", you say that once the configurations are done, you just have to replace the calls to `react-native bundle` by `react-native rnx-bundle`. Except that in a project created by `react-native init`, there is never an explicit call to `react-native bundle`.

So I wonder: is this guide and the features added by rnx-bundle applicable to "classic" react-native projects, created via the `react-native init` command?

Thanks in advance, and thanks again for your work on this tool. 😄

**Comment by tido64** [comment]
I think it just means that where you used to call `react-native bundle ...`, you can call `react-native rnx-bundle ...` instead. All the flags  that are valid for `bundle` are valid for `rnx-bundle` as well. Whether you generated your app from `react-native init` or something else shouldn't matter.

cc @afoxman

**Comment by afoxman** [comment, marked as answer]
Hi @Thanaen, 

Thanks for trying out the guide and taking time to start a discussion! The `rnx-bundle` command is definitely for classic RN apps created with react-native init. It's for any RN app, really, where you use the `react-native bundle`.

I think what's missing here is clear & specific guidance for _new_ `react-native init` projects. When I wrote this, I was trying to address the more complicated scenario of "I have an existing RN app which I bundle, and I want to start using rnx-bundle". I'll get an issue filed to make this distinction -- like maybe under Configuring your App, I'll have sections for New App and Existing App. (Filed as https://github.com/microsoft/rnx-kit/issues/1565).

To help you _now_, @tido64 has the right idea. If you're doing bundling, you can use `rnx-bundle` in place of `bundle`. If you're not creating bundles right now, then you can ignore this and come back to it later. One important note -- `rnx-bundle` isn't quite a drop-in replacement for `bundle` yet. I'm working on a PR to fix this now, and it should be out soon.

This got me thinking -- we should plug into the `react-native init` command so it generates an rnx-kit app, which is really just a regular app with rnx-kit configuration/script-commands. I'm now tracking that work in https://github.com/microsoft/rnx-kit/issues/1263.

**Reply by Thanaen** [reply]
Hi @afoxman, 
Perfect, thanks for your explanations! 👌

**Comment by Thanaen** [comment]
Hello again, @afoxman & @tido64 !

I have a question again after all this time:
After setting up rnx-bundle as shown on this page https://microsoft.github.io/rnx-kit/docs/guides/bundling#setting-up-the-tools, was I supposed to do a "Find and replace" in my project to replace all `react-native start` and `react-native bundle` with `react-native rnx-start` and `react-native rnx-bundle`?

I haven't done it, and I wonder if I've felt like I've been using rnx-bundle for months without actually using it. 😆

**Comment by afoxman** [comment]
Hi @Thanaen, yes, you should use `rnx-bundle` and `rnx-start` everywhere you want to use the rnx-kit CLI.

**Reply by Thanaen** [reply]
Okay, that makes sense. Thank you!
